datafusion/common/src/scalar.rs

```diff
@@ -1011,18 +1011,82 @@ fn ts_sub_to_interval<const TIME_MOD: bool>(
     }
 }
 
-// This function parses the timezone from string to Tz.
-// If it cannot parse or timezone field is [`None`], it returns [`None`].
-pub fn parse_timezones(tz: &Option<String>) -> Option<Tz> {
+/// This function parses the timezone from string to Tz.
+/// If it cannot parse or timezone field is [`None`], it returns [`None`].
+pub fn parse_timezones(tz: &Option<String>) -> Result<Option<Tz>> {
     if let Some(tz) = tz {
-        let parsed_tz: Option<Tz> = FromStr::from_str(tz)
-            .map_err(|_| {
-                DataFusionError::Execution("cannot parse given timezone".to_string())
-            })
-            .ok();
-        parsed_tz
+        let parsed_tz: Tz = FromStr::from_str(tz).map_err(|_| {
+            DataFusionError::Execution("cannot parse given timezone".to_string())
```

Comment:
It would be nice if the error contained the problematic timezone. Something like
```rust
        let parsed_tz: Tz = FromStr::from_str(tz).map_err(|e| {
            DataFusionError::Execution(format!("cannot parse '{tz}' as timezone: {e}".to_string())
```